backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/split/ShipmentScheduleSplitRepository.java

```diff
@@ -132,4 +152,22 @@ public void deleteByIds(@NonNull final Collection<ShipmentScheduleSplitId> ids)
 				.create()
 				.delete(true);
 	}
+
+	public boolean isRelatedInOutValid(@NonNull final I_M_ShipmentSchedule_Split record)
+	{
+		if (record.getM_InOut_ID() <= 0)
+		{
+			return true;
+		}
+		final I_M_InOut inOutRecord = inOutBL.getById(InOutId.ofRepoId(record.getM_InOut_ID()));
+
+		final DocStatus docStatus = DocStatus.ofCode(inOutRecord.getDocStatus());
+
+		return !docStatus.isClosedReversedOrVoided();
+	}
+
+	public boolean isNotProcessed(@NonNull final I_M_ShipmentSchedule_Split record)
```

Comment:
private and consider moving functions below functions where they are used

Reply:
done